src/MarDyn.cpp

```diff
@@ -132,8 +132,18 @@ int run_unit_tests(const Values &options, const std::vector<std::string> &args)
  */
 int main(int argc, char** argv) {
 #ifdef ENABLE_MPI
+#if defined(_OPENMP)
+	int thread_support_level_requested = MPI_THREAD_FUNNELED; /* So far only the main thread makes MPI calls */
+	int thread_support_level_provided = MPI_THREAD_SINGLE;
```

Comment:
I don't quite get why this is set to `MPI_THREAD_SINGLE` if the next line immediately overwrites this value. As far as I understand, `MPI_Init_thread` will always set the last parameter to something, hence, I'd rather suggest default initialization to not give of the impression that the initial value is actually used.

```suggestion
	int thread_support_level_provided{};
```

Reply:
`MPI_THREAD_SINGLE` means "no thread support", therefore make this the initialization to be safe in any case. Note: As long as there is no ABI for MPI (in preparation) you have no guarantees about the integer values for the MPI thread support level constants, i.e., zero could be something unexpected.

Reply:
What do you mean by "in any case"? According to the documentation `MPI_Init_thread` is guaranteed to overwrite this value, hence it shouldn't matter how we initialize it, hence I would argue we shouldn't initialize it to something misleading.
But if you want to keep the value at least add a comment that the value will be overwritten immediately.